internal/domain/favourites/model.go

```diff
@@ -0,0 +1,113 @@
+package favourites
+
+import (
+	"errors"
+	"time"
+)
+
+// AssetType describes the kind of asset stored as a favourite.
+type AssetType string
+
+const (
+	AssetTypeChart    AssetType = "chart"
+	AssetTypeInsight  AssetType = "insight"
+	AssetTypeAudience AssetType = "audience"
+)
+
+// Asset is a tagged union. Type acts as the tag; exactly one of Chart,
+// Insight, or Audience should be non-nil.
+type Asset struct {
+	ID          string
+	Type        AssetType
+	Description string
+
+	Chart    *Chart
+	Insight  *Insight
+	Audience *Audience
```

Comment:
Could you explain why you selected this option? Can you identify any other possible alternatives?

Reply:
I looked at a couple of options here. Go-style polymorphism (an Asset interface with ChartAsset / InsightAsset / AudienceAsset) would definitely work, but for this small service it felt like unnecessary extra complexity and type switches for what is basically a “shape of data” problem.

I ended up with a tagged union instead, as it stays type-safe and keeps the JSON side simple. If this were a bigger system and assets had more behaviours, I’d probably use an Asset interface with concrete types for each asset kind.